### PR TITLE
Transformation sample

### PR DESCRIPTION
**Pull request by dustinlebsock**
The TransformationsClient will NOT work until the api is release to APIM. Scheduled release date is 8/13/2021

### CODE REVIEW

File: src/frontend-samples/transformations-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+The purpose of this sample is to demonstrate the following:
+
+* The usage and result of the Transformation API
+* The logic needed to balance multiple viewports.
+* How to use the Two-way viewport sync feature.
+
+## Description
+
+There are three API's used to create the transformed iModel based on a [View Definition](https://www.itwinjs.org/reference/imodeljs-backend/viewdefinitions/viewdefinition/). The first API and step, [FilterByViewDefinition](https://sbx-developer.bentley.com/api-groups/synchronization/apis/transformations/operations/filterbyviewdefinition/), is to create a transformation configuration which creates an association between source and target iModels with additional metadata to carry out the transformation. It does NOT start the transformation.
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/README.md:17]: Let's update these with prod links: https://developer.bentley.com/apis/transformations/operations/filterbyviewdefinition/

Comment (dsmailys) [reply src/frontend-samples/transformations-sample/README.md:17]: Maybe: "THere are three steps to create the transformed iModel based on a...."

Comment (dsmailys) [reply src/frontend-samples/transformations-sample/README.md:17]: The first step is to create a transformation configuration....

File: src/frontend-samples/transformations-sample/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+* The usage and result of the Transformation API
+* The logic needed to balance multiple viewports.
+* How to use the Two-way viewport sync feature.
+
+## Description
+
+There are three API's used to create the transformed iModel based on a [View Definition](https://www.itwinjs.org/reference/imodeljs-backend/viewdefinitions/viewdefinition/). The first API and step, [FilterByViewDefinition](https://sbx-developer.bentley.com/api-groups/synchronization/apis/transformations/operations/filterbyviewdefinition/), is to create a transformation configuration which creates an association between source and target iModels with additional metadata to carry out the transformation. It does NOT start the transformation. 
+
+The second API, [Create Transformation](https://sbx-developer.bentley.com/api-groups/synchronization/apis/transformations/operations/create-transformation/) actually starts the transformation process by sending in the configuration id.
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/README.md:19]: https://developer.bentley.com/apis/transformations/operations/create-transformation/

File: src/frontend-samples/transformations-sample/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The third API, [Get Transformation](https://sbx-developer.bentley.com/api-groups/synchronization/apis/transformations/operations/get-transformation/), gets an updated transformation status as the transformation is occurring
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/README.md:21]: https://developer.bentley.com/apis/transformations/operations/get-transformation/

File: src/frontend-samples/transformations-sample/TransformationsClient.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { AuthorizedClientRequestContext, IncludePrefix, request, Response } from "@bentley/itwin-client";
+import { AuthorizedFrontendRequestContext,IModelApp } from "@bentley/imodeljs-frontend";
+import { AuthorizationClient } from "@itwinjs-sandbox";
+import { GuidString } from "@bentley/bentleyjs-core";
+
+export interface Transformation {
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/TransformationsClient.ts:10]: This dto changed in APIM a bit, check response dto here: https://developer.bentley.com/apis/transformations/operations/get-transformation/

File: src/frontend-samples/transformations-sample/TransformationsClient.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  finishedDateTime: Date;
+  configurationId: GuidString;
+  sourceChangeSetId: string;
+}
+
+export default class TransformationsClient {
+  private static _requestContext: AuthorizedClientRequestContext;
+
+  public static async getTransformation(transformationId: string): Promise<Transformation | undefined> {
+    const url = `https://sbx-api.bentley.com/transformations/transformations/${transformationId}`;
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/TransformationsClient.ts:26]: https://api.bentley.com/transformations/{transformationId}

File: src/frontend-samples/transformations-sample/walkthru.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Connecting to Transformed iModel
+The purpose of this sample is to demonstrate the Transformation API, however, the sample-showcase is readonly and is unable to create a transformation for the user. Therefore a FilterByViewDefintion Transformation has already been done to the Stadium iModel.
+
+The two lines of code shown is how the sample showcase is used to connect to the Transformed Stadium. GetIModelInfo() grabs the contextId and iModelId for the Transformed Stadium, and then we open a remote connection to connect to.
+
+For more information on the three API's used to accomplish this feat, see:
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/walkthru.md:6]: on the API's used to accomplish

File: src/frontend-samples/transformations-sample/walkthru.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+https://sbx-developer.bentley.com/api-groups/synchronization/apis/transformations/operations/filterbyviewdefinition/
+
```
Comment (dsmailys) [inline src/frontend-samples/transformations-sample/walkthru.md:9]: let's update to prod (already available)

**Review by dsmailys — CHANGES_REQUESTED** [review]
Some changes in regards to wording and api links submitted

File: src/frontend-samples/transformations-sample/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+The purpose of this sample is to demonstrate the following:
+
+* The usage and result of the Transformation API
+* The logic needed to balance multiple viewports.
+* How to use the Two-way viewport sync feature.
+
+## Description
+
+THere are three steps to create the transformed iModel based on a [View Definition](https://www.itwinjs.org/reference/imodeljs-backend/viewdefinitions/viewdefinition/). The first step is to create a transformation configuration, in this example a [FilterByViewDefinition](https://developer.bentley.com/apis/transformations/operations/filterbyviewdefinition/), which creates an association between source and target iModels with additional metadata to carry out the transformation. It does NOT start the transformation.
```
Comment (BaoTon) [inline src/frontend-samples/transformations-sample/README.md:17]:
```suggestion
There are three steps to create the transformed iModel based on a [View Definition](https://www.itwinjs.org/reference/imodeljs-backend/viewdefinitions/viewdefinition/). The first step is to create a transformation configuration, in this example a [FilterByViewDefinition](https://developer.bentley.com/apis/transformations/operations/filterbyviewdefinition/), which creates an association between source and target iModels with additional metadata to carry out the transformation. It does NOT start the transformation. 
```

**Review by BaoTon — COMMENTED** [review]
There seems to be an overlap with the 2d transformations sample when I run it locally:

image

**Review by BaoTon — APPROVED** [review]
LGTM

**Pull request merged**